components/style/selector_matching.rs

```diff
@@ -256,19 +251,18 @@ impl Stylist {
                                         element: &E,
                                         parent_bf: Option<&BloomFilter>,
                                         style_attribute: Option<&PropertyDeclarationBlock>,
-                                        pseudo_element: Option<PseudoElement>,
+                                        pseudo_element: Option<Impl::PseudoElement>,
                                         applicable_declarations: &mut V)
                                         -> bool
-                                        where E: Element + TElement<'le>,
+                                        where E: Element<Impl=Impl> + TElement<'le>,
                                               V: VecLike<DeclarationBlock> {
         assert!(!self.is_device_dirty);
         assert!(style_attribute.is_none() || pseudo_element.is_none(),
                 "Style attributes do not apply to pseudo-elements");
 
         let map = match pseudo_element {
+            Some(ref pseudo) => self.pseudos_map.get(pseudo).unwrap(),
```

Comment:
That's why we pre-insert the eagerly cascaded pseudos map in the stylist constructor, so yes, this can't fail.